src/QsCompiler/Tests.Compiler/CapabilityInferenceTests.fs

```diff
@@ -0,0 +1,63 @@
+﻿module Microsoft.Quantum.QsCompiler.Testing.CapabilityInferenceTests
+
+open Microsoft.Quantum.QsCompiler
+open Microsoft.Quantum.QsCompiler.DataTypes
+open Microsoft.Quantum.QsCompiler.ReservedKeywords.AssemblyConstants
+open Microsoft.Quantum.QsCompiler.SyntaxProcessing
+open Microsoft.Quantum.QsCompiler.SyntaxTree
+open Xunit
+
+/// A mapping of all callables in the capability verification tests, after inferring capabilities.
+let private callables =
+    CompilerTests.Compile ("TestCases", [ "CapabilityVerification.qs" ])
+    |> fun compilation -> compilation.BuiltCompilation
+    |> CapabilityInference.InferCapabilities
+    |> fun compilation -> compilation.Namespaces
+    |> GlobalCallableResolutions
+
+/// Asserts that the inferred capability of the callable with the given name matches the expected capability.
+let private expect capability name =
+    let actual =
+        callables.[CapabilityVerificationTests.testName name].Attributes
+        |> QsNullable<_>.Choose BuiltIn.GetCapability
+    Assert.Equal<RuntimeCapabilities> ([ capability ], actual)
+
+[<Fact>]
+let ``Infers QPRGen0`` () =
```

Comment:
Result comparison for `QPRGen0` not being supported only applies for operations (not for functions)?

Reply:
In QPRGen0, result comparison is forbidden everywhere, including both operations and functions.

Reply:
If you are asking about `ResultTuple` and `ResultArray` being inferred as QPRGen0, it's because at the moment tuples and arrays in Q# do not support equality, so no result equality comparison happens in those functions (they don't compile because equality is not supported). The test case is confusing, though...

Reply:
Thanks for the explanation Sarah! Yes, I was a bit confused about the test case and the only thing I could see different is that it was a function rather than an operation.